cellprofiler/modules/measurecolocalization.py

```diff
@@ -897,53 +1038,25 @@ def get_measurement_images(self, pipeline, object_name, category,
 
     def upgrade_settings(self, setting_values, variable_revision_number,
                          module_name, from_matlab):
-        '''Adjust the setting values for pipelines saved under old revisions'''
-        if from_matlab and variable_revision_number == 3:
-            image_names = [x for x in setting_values[:4]
-                           if x.upper() != cps.DO_NOT_USE.upper()]
-            wants_image_measured = np.any([x == cpmeas.IMAGE
-                                           for x in setting_values[4:]])
-            object_names = [x for x in setting_values[4:]
-                            if not x in (cpmeas.IMAGE, cps.DO_NOT_USE)]
-            if wants_image_measured:
-                if len(object_names):
-                    m = M_IMAGES_AND_OBJECTS
-                else:
-                    m = M_IMAGES
-            elif len(object_names):
-                m = M_OBJECTS
-            else:
-                raise ValueError("Must either measure texture over images or over some set of objects")
-            if len(object_names) == 0:
-                object_names = [cps.NONE]
-            setting_values = ([str(len(image_names)), str(len(object_names))] +
-                              image_names + [m] + object_names)
-            from_matlab = False
-            variable_revision_number = 1
-        if variable_revision_number == 1:
-            #
-            # Wording of image / object text changed
-            #
-            image_count, object_count = [int(x) for x in setting_values[:2]]
-            image_names = setting_values[2:(image_count + 2)]
-            m = setting_values[image_count + 2]
-            object_names = setting_values[(image_count + 3):]
-            if m == "Images":
-                m = M_IMAGES
-            elif m == "Objects":
-                m = M_OBJECTS
-            elif m == "Images and objects":
-                m = M_IMAGES_AND_OBJECTS
-            setting_values = ([str(image_count), str(object_count)] +
-                              image_names + [m] + object_names)
-            variable_revision_number = 2
+        """Adjust the setting values for pipelines saved under old revisions"""
+        if from_matlab:
+            raise NotImplementedError("There is no automatic upgrade path for this module from MatLab pipelines.")
+
+        if variable_revision_number < 2:
+            raise NotImplementedError("Automatic upgrade for this module is not supported in CellProfiler 3.")
+
         if variable_revision_number == 2:
             image_count = int(setting_values[0])
             idx_thr = image_count + 2
             setting_values = \
                 setting_values[:idx_thr] + ["15.0"] + setting_values[idx_thr:]
             variable_revision_number = 3
 
+        if variable_revision_number == 3:
```

Comment:
FWIW I don't think adding a bunch of values on the end of the settings list doesn't need a settings revision. It _should_ just add the defaults (in this case `True`) for all the new settings. I'd try it - setting revision 4 may not be necessary!

Reply:
I can and will check that, but my personal gut feeling is that since we've added a bunch of settings even though we've kept compatibility we should bump the version anyway.  @0x00b1 what do you think there?

Reply:
@AetherUnbound I agree. However, it really depends on how you view the `variable_revision_number`. I think from a user perspective it should be upgraded (but you're right, it shouldn't need to be updated). I suggest @bethac07 (or someone else) comes up with a policy around when to update the `variable_revision_number` that we start to adhere to in future changes. What do you think?

Reply:
I agree with @bethac07 here - from what I understand it's pretty much only for internal use - so if we don't have to change it then I don't think we should.